db/db_impl/db_impl_files.cc

```diff
@@ -252,6 +252,12 @@ void DBImpl::FindObsoleteFiles(JobContext* job_context, bool force,
                                job_context->blob_delete_files);
   }
 
+  // Before potentially releasing mutex and waiting on condvar, increment
+  // pending_purge_obsolete_files_ so that another thread executing
+  // `GetSortedWals` will wait until this thread finishes execution since the
+  // other thread will be waiting for `pending_purge_obsolete_files_`.
```

Comment:
Do you think it would make sense to call out that we have to make sure it is decremented if we end up with nothing to delete (in case someone adds e.g. an early return between here and the decrement below)?

Reply:
Sure. We can utilize the `Defer` type (https://github.com/facebook/rocksdb/blob/main/util/defer.h).